mmv1/templates/terraform/custom_flatten/cloudfunctions2_function_source_bucket.go.erb

```diff
@@ -14,5 +14,8 @@
 # limitations under the License.
 -%>
 func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *Config) interface{} {
-return d.Get("build_config.0.source.0.storage_source.0.bucket")
+    if _, ok := d.GetOk("build_config.0.source.0.storage_source.0.bucket"); ok {
+        return d.Get("build_config.0.source.0.storage_source.0.bucket")
+    }
+    return v
```

Comment:
Will `v` ever be non-empty?

Reply:
Yes, v will be non-empty.
If we keep only d.GetOk(), then it will work for the resource but it won't work for the datasource as it is computed value(i.e. not provided in d) so these attributes are blank in the datasource.
If we keep only v, then it won't work for resource because during refresh state the value of bucket changes to `gcf-v2-****` and the object changes to `{{bucket_name}}/{{object_name}}` so doing the terraform apply again will show changes everytime.
So, I've kept this condition to avoid doing changes for the resource everytime we do terraform apply and also to get these attributes for the datasource as per the Read function.

Reply:
Thanks! Can you write up why the function is written how it is in the code?

Reply:
Yeah, I've added the comments for the same in this [commit](https://github.com/GoogleCloudPlatform/magic-modules/pull/6523/commits/d818a1e8a6fa58abc7089221345e8d65969e6e90)

Reply:
While the comment proposed so far states *what* this does, what's more useful is *why* we're doing it- we're gonna be able to read the code easily enough but the next person to touch the resource isn't likely to be aware of the specific behaviour of the API that causes us to require this customization.

I'd like to include, in the comments of each of these fields:

* What the format returned on GET is, including a comparison of a sample input/output
* That this flattener is shared between the resource (where a prior value is available) and the datasource (where it isn't)

Reply:
Okay, I've updated the comments as suggested in this [commit](https://github.com/GoogleCloudPlatform/magic-modules/pull/6523/commits/4d2a89dc7d043dbbe605577d038600afc9b364f7)